gemfire-core/src/main/java/com/gemstone/gemfire/cache/CacheFactory.java

```diff
@@ -257,7 +258,7 @@ private static Cache basicGetInstance(DistributedSystem system, boolean closeOk)
     // deadlock when messaging returns to this VM
     final int initReq = LocalRegion.threadInitLevelRequirement();
     if (initReq == LocalRegion.ANY_INIT
-        || initReq == LocalRegion.BEFORE_INITIAL_IMAGE) { // fix bug 33471
+        || initReq == LocalRegion.BEFORE_INITIAL_IMAGE || (InitialImageOperation.anyTestHookInstalled())) { // fix bug 33471
```

Comment:
Just bring the fix bug 33471 besides 'initReq == LocalRegion.BEFORE_INITIAL_IMAGE' and put the new condition in the new line